charts/drone/templates/deployment.yaml

```
@@ -29,6 +29,8 @@ spec:
       automountServiceAccountToken: {{ .Values.automountServiceAccountToken }}
       securityContext:
         {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      serviceAccount: {{ .Values.serviceAccount.name }}
+      serviceAccountName: {{ .Values.serviceAccount.name }}
```

```suggestion
      serviceAccountName: {{ .Values.serviceAccount.name }}
```
I think you only need `serviceAccountName`.